nvm.sh

```diff
@@ -1954,6 +1955,51 @@ nvm() {
       fi
       return $?
     ;;
+    "upgrade" )
+      local base_version
+      base_version="$(nvm_ls_current)"
+
+      if [ $# -ge 2 ]; then
+        base_version="$(nvm_match_version $2)"
+
+        if [ $base_version = "N/A" ]; then
+          nvm_err 'Cannot find the specified version'
+          return 1
+        fi
+      fi
+
+      if [ $base_version = "none" ]; then
+        nvm_err 'No version currently active, cannot upgrade'
+        return 1
+      fi
+
+      if [ $base_version = "system" ]; then
+        nvm_err 'Cannot upgrade the system version of node'
+        return 1
+      fi
+
+      local semver_major
+      semver_major="$(echo $base_version | sed 's/v\([0-9]\+\)\.\([0-9]\+\).*/\1/')"
+
+      local semver_minor
+      semver_minor="$(echo $base_version | sed 's/v\([0-9]\+\)\.\([0-9]\+\).*/\2/')"
+
+      local install_version
+      if [ $semver_major = '0' ]; then
+        install_version="v0.$semver_minor"
+      else
+        install_version="v$semver_major"
+      fi
```

Comment:
it'd be great to put this logic in a separate function, like `nvm_get_upgrade_version` or something, so that it can be trivially unit-tested without the need for installation